src/duration.rs

```diff
@@ -401,11 +408,18 @@ impl fmt::Display for Duration {
 pub struct OutOfRangeError(());
 
 impl fmt::Display for OutOfRangeError {
+    #[cfg(feature = "std")]
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         write!(f, "{}", self.description())
     }
+
+    #[cfg(not(feature = "std"))]
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f,"Source duration value is out of range for the target type")
```

Comment:
We run into borrow-checker issues here as Error wants a `&str` but `Display/ToString` allocates a String. I've replaced the duplication with a const.